woocommerce/woocommerce
# [CYS - Core] Before installing the fonts, ensure that the font library is available

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces a check to ensure that the Font Library is installed before proceeding with the font installation.

Closes #44315 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new WooCommerce installation with this version.
2. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
3. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
4. Install this build of Gutenberg that contains the new Font Library API: 
[gutenberg.zip](https://github.com/woocommerce/woocommerce/files/14085546/gutenberg.zip)
5. Visit the `wp-admin/admin.php?page=wc-admin&path=/customize-store`.
6. Open the network tools.
7. Follow the process.
8. Ensure that there are some requests to `/font-face` and `/font-families` endpoints.
9. Uninstall Gutenberg.
10. Visit the `wp-admin/admin.php?page=wc-admin&path=/customize-store`.
11. Follow the process.
12. Open the network tools.
13. Ensure there are no request to `/font-face` and `/font-families` endpoints.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 
[CYS - Core]: Before installing the fonts, ensure that the font library is available


#### Comment 

</details>
